api/v1/articles/ArticlesHandler.inc.php

```diff
@@ -0,0 +1,175 @@
+<?php 
+
+/**
+ * @file api/v1/articles/ArticlesHandler.inc.php
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ArticlesHandler
+ * @ingroup api_v1_articles
+ *
+ * @brief Handle API requests for articles operations.
+ *
+ */
+
+import('lib.pkp.classes.handler.APIHandler');
+import('classes.core.ServicesContainer');
+
+class ArticlesHandler extends APIHandler {
+	
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		$this->_handlerPath = 'articles';
+		$roles = array(ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_ASSISTANT, ROLE_ID_REVIEWER, ROLE_ID_AUTHOR);
+		$this->_endpoints = array(
+			'GET' => array (
+					array(
+						'pattern' => $this->getEndpointPattern() . '/{articleId}',
+						'handler' => array($this,'getArticle'),
+						'roles' => $roles
+					),
+				)
+		);
+		parent::__construct();
+	}
+	
+	//
+	// Implement methods from PKPHandler
+	//
+	function authorize($request, &$args, $roleAssignments) {
+		$routeName = null;
+		$slimRequest = $this->getSlimRequest();
+	
+		if (!is_null($slimRequest) && ($route = $slimRequest->getAttribute('route'))) {
+			$routeName = $route->getName();
+		}
+		
+		import('lib.pkp.classes.security.authorization.ContextRequiredPolicy');
+		$this->addPolicy(new ContextRequiredPolicy($request));
+		
+		import('classes.security.authorization.OjsJournalMustPublishPolicy');
+		$this->addPolicy(new OjsJournalMustPublishPolicy($request));
+		
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+	
+	//
+	// Public handler methods
+	//
+	/**
+	 * Handle article view
+	 * @param $slimRequest Request Slim request object
+	 * @param $response Response object
+	 * @param array $args arguments
+	 * @return Response
+	 */
+	public function getArticle($slimRequest, $response, $args) {
+		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_READER, LOCALE_COMPONENT_PKP_SUBMISSION);
+		
+		$request = $this->getRequest();
+		$dispatcher = $request->getDispatcher();
+		$context = $request->getContext();
+		$journal = $request->getJournal();
+
+		$articleId = $this->getParameter('articleId');
+		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+		$publishedArticle = $publishedArticleDao->getPublishedArticleByBestArticleId((int) $journal->getId(), $articleId, true);
+		
+		$issue = null;
+		$article = null;
+		$issueDao = DAORegistry::getDAO('IssueDAO');
+		if (isset($publishedArticle)) {
+			$issue = $issueDao->getById($publishedArticle->getIssueId(), $publishedArticle->getJournalId(), true);
+			$article = $publishedArticle;
+		} else {
+			$articleDao = DAORegistry::getDAO('ArticleDAO');
+			$article = $articleDao->getById((int) $articleId, $journal->getId(), true);
```

Comment:
At a glance, this bypasses any authorization checks -- anonymous users could request e.g. unpublished content. This is where the authorization framework needs to be knit in, I think. The policies themselves should fetch the article object and put it in the authorized context, and this code should just need to add the right policy and then fetch the object from the policy where it should already live.